src/Grid/grid.jl

```diff
@@ -823,3 +824,81 @@ struct SurfaceOrientationInfo
     #flipped::Bool
     #shift_index::Int
 end
+
+
+@doc raw"""
+    InterfaceTransformation
+
+Orientation information for 1D and 2D interfaces in 2D and 3D elements respectively. 
+This information is used to construct the transformation matrix to
+transform the quadrature points from face_a to face_b achieving synced
+spatial coordinates. Face B's orientation relative to Face A's can
+possibly flipped (i.e. the vertices indices order is reversed)
+and the vertices can be rotated against each other. 
+The reference orientation of face B is such that the first node
+has the lowest vertex index. Thus, this structure also stores the
+shift of the lowest vertex index which is used to reorient the face in
+case of flipping ["transform_interface_point!"](@ref).
+Take for example the faces
+```
+1           2
+| \         | \
+|  \        |  \
+| A \       | B \ 
+|    \      |    \
+2-----3     3-----1  
+```
+which are rotated against each other by 240° after tranfroming to an
+equilateral triangle (shift index is 2) or the faces
+```
+2           2
+| \         | \
+|  \        |  \
+| A \       | B \ 
+|    \      |    \
+3-----1     3-----1  
+```
+which are flipped against each other, note that face B has its reference node shifted by 2 indices
+so the face is tranformed into an equilateral triangle then rotated 120°, flipped about the x axis then
+rotated -120° and tranformed back to the reference triangle.Any combination of these can happen. 
+"""
+struct InterfaceTransformation{nfacenodes}
+    flipped::Bool
+    shift_index::Int
+    lowest_node_shift_index::Int
+    # Used to dispatch the correct 2/3D -> 3/2D Transformations
+    cell_a_refshape::Type{<:AbstractRefShape}
+    face_a_index::Int
+    cell_b_refshape::Type{<:AbstractRefShape}
```

Comment:
Can the two refshapes be type parameters here instead and remove the `nfacenodes` one?